sql_it_test.go

```diff
@@ -0,0 +1,581 @@
+		if err != nil {
+			t.Fatal(err)
+		}
+		var vs []interface{}
+		var k int64
+		var v interface{}
+		if err := row.Scan(&k, &v); err != nil {
+			t.Fatal(err)
+		}
+		vs = append(vs, v)
+		targetThis := []interface{}{&Record{
+			VarcharValue:  "hello",
+			TinyIntValue:  -128,
+			SmallIntValue: 32767,
+			IntegerValue:  -27,
+			BigIntValue:   38,
+			BoolValue:     true,
+			RealValue:     -5.32,
+			DoubleValue:   12.789,
+			DecimalValue:  &dec,
+		}}
+		assert.Equal(t, targetThis, vs)
+		// select individual fields
+		row, err = queryRow(client, fmt.Sprintf(`
+			SELECT __key, varcharvalue, tinyintvalue, smallintvalue, integervalue,	
+			bigintvalue, boolvalue, realvalue, doublevalue, decimalvalue from "%s"`, mapName))
+		if err != nil {
+			log.Fatal(err)
+		}
+		vs = nil
+		var vVarchar string
+		var vTinyInt int8
+		var vSmallInt int16
+		var vInteger int32
+		var vBigInt int64
+		var vBool bool
+		var vReal float32
+		var vDouble float64
+		var vDecimal types.Decimal
+		if err := row.Scan(&k, &vVarchar, &vTinyInt, &vSmallInt, &vInteger, &vBigInt, &vBool, &vReal, &vDouble, &vDecimal); err != nil {
+			t.Fatal(err)
+		}
+		vs = append(vs, vVarchar, vTinyInt, vSmallInt, vInteger, vBigInt, vBool, vReal, vDouble, vDecimal)
+		target := []interface{}{
+			"hello", int8(-128), int16(32767), int32(-27), int64(38), true, float32(-5.32), 12.789, types.NewDecimal(big.NewInt(123_456_789), 100),
+		}
+		assert.Equal(t, target, vs)
+	})
+}
+
+func TestSQLWithPortableDateTime(t *testing.T) {
+	it.SkipIf(t, "hz < 5.0")
+	cb := func(c *hz.Config) {
+		c.Serialization.SetPortableFactories(&recordFactory{})
+	}
+	it.SQLTesterWithConfigBuilder(t, cb, func(t *testing.T, client *hz.Client, config *hz.Config, m *hz.Map, mapName string) {
+		q := fmt.Sprintf(`
+			CREATE MAPPING "%s" (
+				__key BIGINT,
+				datevalue DATE,
+				timevalue TIME,
+				timestampvalue TIMESTAMP,
+				timestampwithtimezonevalue TIMESTAMP WITH TIME ZONE
+			)
+			TYPE IMAP
+			OPTIONS (
+				'keyFormat' = 'bigint',
+				'valueFormat' = 'portable',
+				'valuePortableFactoryId' = '100',
+				'valuePortableClassId' = '2'
+			)
+		`, mapName)
+		t.Logf("Query: %s", q)
+		it.MustValue(client.ExecSQL(context.Background(), q))
+		dt := time.Date(2021, 12, 22, 23, 40, 12, 3400, time.FixedZone("A/B", -5*60*60))
+		rec := NewRecordWithDateTime(&dt)
+		_, err := client.ExecSQL(context.Background(), fmt.Sprintf(`INSERT INTO "%s" (__key, datevalue, timevalue, timestampvalue, timestampwithtimezonevalue) VALUES(?, ?, ?, ?, ?)`, mapName),
+			1, *rec.DateValue, *rec.TimeValue, *rec.TimestampValue, *rec.TimestampWithTimezoneValue)
+		if err != nil {
+			t.Fatal(err)
+		}
+		targetDate := time.Date(2021, 12, 22, 0, 0, 0, 0, time.Local)
+		targetTime := time.Date(0, 1, 1, 23, 40, 12, 3400, time.Local)
+		targetTimestamp := time.Date(2021, 12, 22, 23, 40, 12, 3400, time.Local)
+		targetTimestampWithTimezone := time.Date(2021, 12, 22, 23, 40, 12, 3400, time.FixedZone("", -5*60*60))
```

Comment:
Could you clarify your point?

Reply:
I am talking about the string in time.FixedZone. We pass "A/B" while creating the record but we assert it to be empty string. I was just wondering why. Also what does A/B mean?

Reply:
A/B has no meaning. The codec for the timestamps does not support location names, so that information cannot be restored, hence it is empty when decoded.